operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/model/url/DownloadUrl.java

```diff
@@ -119,10 +123,15 @@ public String getSignedUrl() throws Exception {
                 "&X-Amz-Expires=" + expirySeconds +
                 "&X-Amz-SignedHeaders=host";
 
-        String standardizedResource = "/" + bucketName + "/" + cosBucketPathPrefix + "/"+ objectKey;
+        // VirtualHost/Path - we inline the bucket if it's path based access.
+        String bucketResource = "";
+        if (path) {
+            bucketResource = "/" + bucketName;
+        }
+        String standardizedResource = bucketResource + "/" + cosBucketPathPrefix + "/"+ objectKey;
 
         String payloadHash = "UNSIGNED-PAYLOAD";
-        String standardizedHeaders = "host:" + server.replaceAll("https://", "");
+        String standardizedHeaders = "host:" + server.replace("https://", "");
```

Comment:
is that variable used somewhere?

Reply:
Yes - it's used to recalculate the presigned URL